service/history/ndc/workflow_resetter_test.go

```diff
@@ -915,6 +916,73 @@ func (s *workflowResetterSuite) TestReapplyEvents() {
 	_, err = reapplyEvents(context.Background(), ms, nil, smReg, events, nil, "")
 	s.NoError(err)
 }
+func (s *workflowResetterSuite) TestReapplyEvents_Excludes() {
```

Comment:
I would much rather have a functional test for this. There's already a reset test in `nexus_workflow_test.go`, please add one there.

Reply:
Added.
While testing I noticed that if we reset the workflow to a point before `EVENT_TYPE_NEXUS_OPERATION_SCHEDULED`, the whole set of nexus operations will be excluded by default? Looks like this is happening because `EVENT_TYPE_NEXUS_OPERATION_SCHEDULED` cannot be `CherryPick()`'d. And so the subsequent nexus started & completed operations also cannot be cherrypicked since the state machine won't be found. Is that intentional?

Reply:
Yes, all of the other nexus events have a reference to the scheduled event. They can't be cherry picked without that base event.

Reply:
So, should we make the schedule event cherry pickable as well then? Without that, we cannot reset workflow before the scheduled event.

Reply:
Schedule event can't be cherry-pickable, it's a workflow command, only workflow logic can generate this. It's the same for scheduling an activity or a child workflow.

It doesn't make sense to cherry pick any events that depend on the scheduled event, what we have is fine.
We would still need other fixes to be able to re-attach to the same workflow if we reset to before the scheduled command but that's a separate issue.